tools/caffe.cpp

```diff
@@ -195,6 +195,7 @@ int train() {
   shared_ptr<caffe::Solver<float> >
     solver(caffe::GetSolver<float>(solver_param));
 
+  solver->CheckSnapshotWritePermissions();
```

Comment:
Maybe put this check into `Solver::Init` instead of here, to also help pycaffe and matcaffe?